### PR TITLE
PVP-41

### PR DESCRIPTION
**Pull request by KhajiitLikeToSneak**
Added filters functionality
Design could be updated later on

### CODE REVIEW

File: app/src/main/java/com/pvp/app/ui/screen/filter/FilterViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                .getCurrent()
+                .first()
+        }
+    }
+
+    fun updateUserActivities(activities: List<String>) {
+        viewModelScope.launch {
+            val currentUser = user.value ?: return@launch
+            val updatedUser = currentUser.copy(activities = activities)
+            userService.merge(updatedUser)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/filter/FilterViewModel.kt:34]: Separation from variables initialization is required.

File: app/src/main/java/com/pvp/app/ui/screen/filter/FilterViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            _user.emit(updatedUser)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/filter/FilterViewModel.kt:35]: Separation required.

**Review by klaidoshka — CHANGES_REQUESTED** [review]

**Review by vytjar — APPROVED** [review]
Functionality seems okay. Will need a pop up or something in the future to inform the user that changes have been applied.

File: app/src/main/java/com/pvp/app/ui/screen/filter/Filter.kt [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    ) {
+        Column(
+            modifier = Modifier.fillMaxWidth()
+        ) {
+            Text(
+                text = title,
+                style = TextStyle(fontSize = 18.sp),
+                modifier = Modifier.padding(8.dp)
+            )
+            FlowRow(
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/filter/Filter.kt:105]: Empty single line could be added to separate components of the same level.

File: app/src/main/java/com/pvp/app/ui/screen/filter/Filter.kt [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                                MaterialTheme.shapes.medium
+                            ),
+                        onClick = { onClick(activity) }
+                    ) {
+                        Text(
+                            text = activity,
+                            style = TextStyle(fontSize = 16.sp),
+                            modifier = Modifier.padding(16.dp)
+                        )
+                    }
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/filter/Filter.kt:127]: Whole card with its children components may not be correctly indented. `Control+Alt+L` would fix this. I think 4 spaces are missing.

File: app/src/main/java/com/pvp/app/ui/screen/filter/Filter.kt [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+import androidx.compose.material3.Text
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.compose.runtime.LaunchedEffect
+import com.pvp.app.model.SportActivity
+
+@Composable
+fun ActivitiesFilter(
+    model: FilterViewModel = hiltViewModel()
+) {
+    val activities: List<String> = SportActivity.values().map { it.title }
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/filter/Filter.kt:35]: Could be transfered out of the activities filter component to make it singleton (computed only once) or into `LaunchedEffect` block to make it recompute only after user value changes, but not every time the component is rerendered.

**Review by klaidoshka — APPROVED** [review]

**Pull request merged**